packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1753,7 +1742,7 @@ class StackViewport extends Viewport implements IStackViewport, IImagesLoader {
       currentImageIdIndex: currentImageIdIndex,
     };
 
-    triggerEvent(eventTarget, Events.STACK_VIEWPORT_NEW_STACK, eventDetail);
+    triggerEvent(this.element, Events.STACK_VIEWPORT_NEW_STACK, eventDetail);
```

Comment:
Could you make this a generic event, rather than being stack specific?  SOmething like VIEWPORT_NEW_IMAGE_SET
It is really annoying to have to figure out which set of events to listen for.